docs/adr/2023-06-19_024-persistence.md

```diff
@@ -21,35 +21,44 @@ Proposed
+    activate State
+    State -->> Node : state: HeadState
+    deactivate State
+    Node ->> HeadLogic: update(state, event)
+    activate HeadLogic
+    HeadLogic -->> Node : Combined (sc: StateChanged) (oe: OtherEffect)
+    deactivate HeadLogic
+    Node ->> Node: aggregateState(state, sc)
+    Node ->> State: setState(state)
+    activate State
+    State ->> Node: ok
+    deactivate State
+   end
+   Node ->> Node: append(sc)
+   Node -) HandleOtherEffect: processEffect(oe)
 ```
 
 ## Consequences
 
--  :racehorse: The main expected consequence of this change is an increase of the overall performance of Hydra Head network
--  While _Effect handlers_ might be asynchronous, `StateChanged` handler _must_ be synchronous to ensure consistency and durability of state changes
+-  :racehorse: The main expected consequence of this change is an increase of the overall performance of Hydra Head network.
+-  While _Effect handlers_ might be asynchronous, `StateChanged` handler _must_ be synchronous to ensure consistency and durability of state changes.
 - An longer term consequence is the possibilities this change introduces with respect to `ServerOutput` handling and client's access to a head's state:
-  - Instead of having the `HeadLogic` emits directly a `ClientEffect`, the latter could be the result of a client-centric _interpretation_ of a `StateChanged`
-  - Pushing this a little further, we could maintain a _Query Model_ for clients with a dedicated [Query API](https://github.com/input-output-hk/hydra/discussions/686) to ease implementation of stateless clients
-- Crashing nodes could catch-up with the Head's state by requesting `StateChanged` changes they are missing
-- Calling `StateChanged` an _event_ while treating it in the code alongside _effects_ might introduce some confusion as we already use the word [Event](https://github.com/input-output-hk/hydra/blob/45913954eb18ef550a31017daa443cee6720a00c/hydra-node/src/Hydra/HeadLogic.hs#L64) to designate the inputs to the Head logic state machine. We might want at some later point to unify the terminology
+  - Instead of having the `HeadLogic` emits directly a `ClientEffect`, the latter could be the result of a client-centric _interpretation_ of a `StateChanged`.
+  - Pushing this a little further, we could maintain a _Query Model_ for clients with a dedicated [Query API](https://github.com/input-output-hk/hydra/discussions/686) to ease implementation of stateless clients.
+- Crashing nodes could catch-up with the Head's state by requesting `StateChanged` changes they are missing.
+- Calling `StateChanged` an _event_ while treating it in the code alongside _effects_ might introduce some confusion as we already use the word [Event](https://github.com/input-output-hk/hydra/blob/45913954eb18ef550a31017daa443cee6720a00c/hydra-node/src/Hydra/HeadLogic.hs#L64) to designate the inputs to the Head logic state machine. We might want at some later point to unify the terminology.
```

Comment:
This is missing one consequence/drawbacks we have identified throughout executing this:
* Terms from specification are distributed over `update` and `aggregate` functions